Admin/DynamicListAdmin.php

```diff
@@ -45,20 +45,27 @@ public function __construct(ViewBuilderFactoryInterface $viewBuilderFactory, arr
 
     public function configureViews(ViewCollection $viewCollection): void
     {
-        $action = 'form-list';
-        // TODO handle multipage views
-
         foreach ($this->config as $parent => $sections) {
-            foreach ($sections as $key => $config) {
+            /** @var mixed[] $config */
+            foreach ($sections as $config) {
                 if (!$viewCollection->has($parent)) {
                     continue;
                 }
 
-                $name = $parent . '.' . $action . '-key';
+                $action = 'form-list';
+                if (isset($config['action'])) {
+                    $action = $config['action'];
+                }
```

Comment:
Following way we avoid the bc break when using the key:

```suggestion
                $action = 'form-list';
                if ($counter > 1) {
                    $action .= '-' . str_replace('_', '-', $key);
                }
```